state-sync/storage-service/server/src/lib.rs

```diff
@@ -542,6 +661,15 @@ impl StorageReaderInterface for StorageReader {
         end_account_index: u64,
     ) -> Result<StateValueChunkWithProof, Error> {
         let expected_num_accounts = inclusive_range_len(start_account_index, end_account_index)?;
+        let max_account_chunk_size = self.config.max_account_states_chunk_sizes;
+        if expected_num_accounts > max_account_chunk_size {
+            return Err(Error::InvalidRequest(format!(
+                "Requested number of accounts is larger than the maximum! \
+             Requested: {:?}, maximum: {:?}.",
+                expected_num_accounts, max_account_chunk_size
+            )));
+        }
```

Comment:
why not take min(expected, max)?

Reply:
Because state sync v2 considers that an invalid response. The driver decides what data sizes to ask for (based on data availability around it). If the driver asks for 10 txns and you give it only 5, it'll discard it.